frontend/components/portfolio/group/membership/MenuButton.vue

```diff
@@ -0,0 +1,175 @@
+    }
+    emit('changed-memberships')
+    toast.add({
+      severity: 'success',
+      group: add ? 'portfolio-group-membership-added' : 'portfolio-group-membership-removed',
+    })
+    // toggleMenu(event)
+  }
+}
+const changedGroups = () => {
+  emit('changed-groups')
+}
+
+type Icon = 'empty' | 'partial' | 'full'
+
+const groupOptions = computed(() => {
+  const selected = props.selectedPortfolios
+  const result = props.portfolioGroups.map((pg) => {
+    const isMember = selected.map((portfolio) => (portfolio.groups ?? []).some((pg2) => pg.id === pg2.portfolioGroup.id))
+    const anySelected = isMember.some(m => m)
+    const allSelected = isMember.every(m => m)
+    let icon: Icon = 'empty'
+    let addIfClick = true
+    let hoverText = tt('Add all portfolios to group')
+    if (allSelected) {
+      icon = 'full'
+      addIfClick = false
+      hoverText = tt('Remove all portfolios from group')
+    } else if (anySelected) {
+      // TODO(grady) make a pi-square-minus
+      icon = 'partial'
+      addIfClick = true
+      hoverText = tt('Add unselected portfolios to group')
+    }
+    return {
+      id: pg.id,
+      label: pg.name,
+      icon,
+      cmd: changeMemberships(pg.id, addIfClick),
+      hoverText,
+      created: pg.createdAt,
+    }
+  })
+  result.sort((a, b) => a.created < b.created ? 1 : -1)
+  return result
+})
+</script>
+
+<template>
+  <div>
+    <PVButton
+      v-if="props.selectedPortfolios && props.selectedPortfolios.length > 0"
+      :class="visible ? '' : 'p-button-outlined'"
+      :label="tt('Group Memberships')"
+      icon="pi pi-sitemap"
+      @click="toggleMenu"
+    />
+    <PVOverlayPanel
+      ref="overlayPanel"
+      :pt="{ content: { class: 'p-0' } }"
+      @hide="hideMenu"
+      @show="showMenu"
+    >
+      <div class="flex flex-column align-items-stretch">
+        <div class="font-bold text-xl p-3 border-bottom-1 border-600 flex gap-2 align-items-center">
+          <span>{{ tt('Group Memberships') }}</span>
+          <PVButton
+            icon="pi pi-times"
+            class="p-button-text px-1 py-0 w-auto h-auto p-button-secondary"
+            @click="toggleMenu"
+          />
+        </div>
+        <div
+          v-for="option in groupOptions"
+          :key="option.id"
+          class="border-bottom-1 border-400"
+        >
+          <PVButton
+            v-tooltip="option.hoverText"
+            class="text-left p-button-text w-full"
+            @click="option.cmd"
+          >
+            <div class="flex justify-content-start align-items-center gap-3">
+              <div
+                class="flex-0 border-2 border-round flex justify-content-center align-items-center"
+                :class="option.icon === 'empty' ? 'bg-white' : 'bg-primary-500 text-white border-primary-500'"
+                style="height: 1.25rem; width: 1.25rem;"
```

Comment:
lol `h-1rem` and `w-1rem` weren't doing it for you?

But more seriously, if these values are useful elsewhere (e.g. for consistency), consider putting them in some utility class or variables in the top level SCSS/CSS.

Reply:
SG - I'll put them elsewhere.

Yeah, my goal was to match the size and vibe of the checkboxes that PV uses, but since we include a third state (dash) for mixed uses, PV didn't have a solution, so I built this.